spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/JdbcAggregateOperations.java

```diff
@@ -50,6 +53,8 @@ public interface JdbcAggregateOperations {
 	 * @param instances the aggregate roots to be saved. Must not be {@code null}.
 	 * @param <T> the type of the aggregate root.
 	 * @return the saved instances.
+	 * @throws IncorrectUpdateSemanticsDataAccessException when at least one instance is determined to be not new and the
```

Comment:
Would `JdbcUpdateAffectedIncorrectNumberOfRowsException` be the right one to use?

Reply:
Probably yes.

Reply:
We still can define `IncorrectUpdateSemanticsDataAccessException` here, I was just wondering that we have no code changes associed.

Reply:
I just looked into changing the actual exception thrown into a `JdbcUpdateAffectedIncorrectNumberOfRowsException`. It would require passing a SQL statement and an expected and actual number of updated rows to the constructor. This seems like the wrong abstraction level to me and I'd rather keep the `IncorrectUpdateSemanticsDataAccessException`.